misc-scripts/xref_mapping/XrefParser/RefSeqGPFFParser.pm

```diff
@@ -17,340 +17,330 @@ limitations under the License.
-
-
-  $peptide_source_id =
-    $self->get_source_id_for_source_name('RefSeq_peptide', undef, $dbi);
-  $mrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_mRNA','refseq', $dbi);
-  $ncrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_ncRNA', undef, $dbi);
-
-  $pred_peptide_source_id =
-    $self->get_source_id_for_source_name('RefSeq_peptide_predicted', undef, $dbi);
-  $pred_mrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_mRNA_predicted','refseq', $dbi);
-  $pred_ncrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_ncRNA_predicted', undef, $dbi);
-
-  $entrez_source_id = $self->get_source_id_for_source_name('EntrezGene', undef, $dbi);
-  $wiki_source_id = $self->get_source_id_for_source_name('WikiGene', undef, $dbi);
-
-  if($verbose){
-    print "RefSeq_peptide source ID = $peptide_source_id\n";
-    print "RefSeq_mRNA source ID = $mrna_source_id\n";
-    print "RefSeq_ncRNA source ID = $ncrna_source_id\n";
-    print "RefSeq_peptide_predicted source ID = $pred_peptide_source_id\n";
-    print "RefSeq_mRNA_predicted source ID = $pred_mrna_source_id\n" ;
-    print "RefSeq_ncRNA_predicted source ID = $pred_ncrna_source_id\n" ;
+
+  $self->{species_id} = $species_id;
+  $self->{dbi} = $dbi;
+  $self->{verbose} = $verbose;
+
+  # get RefSeq source ids
+  while (my ($source_prefix, $source_name) = each %{$REFSEQ_SOURCES}) {
+    $self->{source_ids}->{$source_name} = $self->get_source_id_for_source_name( $source_name, undef, $dbi )
   }
 
-  (%entrez)     = %{$self->get_acc_to_label("EntrezGene",$species_id, undef, $dbi)};
+  # get extra source ids
+  $self->{source_ids}->{EntrezGene} = $self->get_source_id_for_source_name('EntrezGene', undef, $dbi);
+  $self->{source_ids}->{WikiGene} = $self->get_source_id_for_source_name('WikiGene', undef, $dbi);
+
+  # Retrieve existing RefSeq mRNA
+  $self->{refseq_ids} = { %{$self->get_valid_codes('RefSeq_mRNA', $species_id, $dbi)},
+      %{$self->get_valid_codes('RefSeq_mRNA_predicted', $species_id, $dbi)} };
+  $self->{entrez_ids} = $self->get_valid_codes('EntrezGene', $species_id, $dbi);
+  $self->{wiki_ids} = $self->get_valid_codes('WikiGene', $species_id, $dbi);
+
+  if ($verbose) {
+    for my $source_name (sort values %{$REFSEQ_SOURCES}) {
+      print "$source_name source ID = $self->{source_ids}->{$source_name}\n";
+    }
+  }
+
+  # populate entrez gene id => label hash
+  $self->{entrez} = $self->get_acc_to_label('EntrezGene', $species_id, undef, $dbi);
+
+  # get the species name, prepare species related data checks
+  my %species2name = $self->species_id2name($dbi);
+  $species_name //= shift @{$species2name{$species_id}};
+
+  $self->{name2species_id} = { map{ $_=>$species_id } @{$species2name{$species_id}} };
+  my %species2tax  = $self->species_id2taxonomy($dbi);
+  $self->{taxonomy2species_id} = { map{ $_=>$species_id } @{$species2tax{$species_id}} };
+
+  # process the source files
+  foreach my $file (@{$files}) {
```

Comment:
Given there are nested loops in this parser, it really wouldn't hurt to label all of them.